Template/Variable/CustomRequestProcessingVariable.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\TagManager\Template\Variable;
+
+use Piwik\Piwik;
+use Piwik\Settings\FieldConfig;
+use Piwik\Validators\NotEmpty;
+
+class CustomRequestProcessingVariable extends BaseVariable
+{
+    const ID = 'CustomRequestProcessing';
+
+    public function getId()
+    {
+        return self::ID;
+    }
+
+    public function getCategory()
+    {
+        return self::CATEGORY_OTHERS;
+    }
+
+    public function isCustomTemplate()
+    {
+        return true;
+    }
+
+    public function getParameters()
+    {
+        return array(
+            $this->makeSetting('jsFunction', "function(request){\n\treturn request;\n}", FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = Piwik::translate('TagManager_CustomRequestProcessingVariableJsFunctionTitle');
+                $field->description = Piwik::translate('TagManager_CustomRequestProcessingVariableJsFunctionDescription');
+                $field->uiControl = FieldConfig::UI_CONTROL_TEXTAREA;
+                $field->validators[] = new NotEmpty();
+                $field->validate = function ($value) {
+                    $value = trim($value);
+                    if (strpos($value, 'function(request){') !== 0) {
+                        throw new \Exception('The value needs to start with "function(request){"');
```

Comment:
Since this is a validation error that will be surfaced to the customer, I wonder if it should be translated?

Reply:
I wanted it consistent with the existing code here: https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Variable/CustomJsFunctionVariable.php#L44

so i also did not use translation. I have no problem using translation, but then I'd like to change it in `CustomJsFunctionVariable.php`, too. Just let me know what you prefer 😄

Reply:
So I double checked an noticed that exception translation is not consistent.

Regarding Variables, they are (in addition to `CustomJsFunctionVariable.php` as above) not translated in
https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Variable/DomElementVariable.php#L41
https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Variable/DomElementVariable.php#L51
https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Variable/EtrackerConfigurationVariable.php#L128

On the other hand, they are translated in the [`MatomoConfigurationVariable`](
https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Variable/MatomoConfigurationVariable.php).

Regarding tags and triggers they seem to be never translated. ([`GoogleAnalyticsUniversalTag.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Tag/GoogleAnalyticsUniversalTag.php), [`ShareaholicTag.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Tag/ShareaholicTag.php),  [`TagsProvider.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Tag/TagsProvider.php), [`ThemeColorTag.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Tag/ThemeColorTag.php), [`ZendeskChatTag.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Tag/ZendeskChatTag.php), [`ElementVisibilityTrigger.php`](https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Trigger/ElementVisibilityTrigger.php), [`TriggersProvider.php`](
https://github.com/matomo-org/tag-manager/blob/e3c5a5ecd5b1e968f2bba98eaf1e19909ac78ebf/Template/Trigger/TriggersProvider.php))

Reply:
We'll probably want to translate those at some point, but it's fine as they are for now :+1: